modAionImpl/src/org/aion/zero/impl/blockchain/AionBlockchainImpl.java

```diff
@@ -1598,61 +1601,58 @@ public boolean isValid(BlockHeader header) {
             return false;
         }
 
-        Block parent = getParent(header);
-        if (parent == null) {
+        Block[] threeGenParents = getBlockStore().getThreeGenerationBlocksByHashWithInfo(header.getParentHash());
+
+        if (threeGenParents == null) {
             return false;
         }
 
-        Block grandParent = getParent(parent.getHeader());
+        Block parentBlock = threeGenParents[0];
+        Block greatParentBlock = threeGenParents[1];
```

Comment:
minor: typo in names, should have been `grandparentBlock` & `greatGrandparentBlock`